packages/compass-app-stores/src/stores/instance-store.js

```diff
@@ -60,7 +60,10 @@ store.fetchDatabaseDetails = async(dbName, { nameOnly = false } = {}) => {
   const { instance, dataService } = store.getState();
   const db = instance.databases.get(dbName);
 
-  if (db.collectionsStatus === 'initial') {
+  if (
+    db.collectionsStatus === 'initial' ||
+    db.collectionsStatus === 'fetching'
```

Comment:
> Small thing, but how do you feel about having these states as constants/an enum we can refer to?

I would like to, but I legit don't know where to put them and I don't want to move the models in one package at least in the scope of this work if that's okay (I might do this as a separate PR or something, there is enough reasons for that already, sounds alright?)

> Could we potentially be redoing the same query many times?

All inflight requests for the model data are currently debounced (for all the instance related models, but I'll link only to one code example), so if multiple parts of the application are calling `instance.fetch`, they will be awaiting on the same request promise:

https://github.com/mongodb-js/compass/blob/a49978800cfefbd05b71e93b0671adf67a960b24/packages/instance-model/lib/model.js#L6-L35

This particular check here doesn't mean that we start another fetch while the current one is in progress, it just guarantees we wait for the in-progress one to finish before starting to fetch collection info.

Giving it another look I am thinking that I'll probably move this to the models, similar how the whole `instance.refresh` flow is now part of the model so that the stores are not doing much except for just calling model methods directly

Reply:
Done in 5548eec04ba6dc207dc109c2a8e85fb02b1c9e91